src/nrnpython/nrnpy_hoc.cpp

```diff
@@ -1571,6 +1571,7 @@ static PyObject* iternext_sl(PyHocObject* po, hoc_Item* ql) {
     for (;;) {  // have to watch out for deleted sections.
       hoc_Item* q1 = q->next;
       Section* sec = q->element.sec;
+      if (sec == NULL) return NULL;
```

Comment:
I'm not sure this is correct. Shouldn't you  remove q from the list and continue the iteration. Did you test with
three sections and delete the middle one?

Reply:
There is not a problem with 3 sections. I did try including a call to hoc_l_delete(q), but this lead to a segfault at nrnpy_nrn.cpp:741 where sec and sec->prop are both accessible, but sec->prop->dparam is not.

Reply:
I'm going to delve more deeply into this. When I run with the pyiter branch (without this change) I get
```
Assertion failed: file /home/hines/neuron/temp/src/nrnoc/solve.c, line 543
nrniv: !sec->parentsec
```
Generally a 'delete dend' would only destroy the dend immediately if there the reference count for the hoc Section
goes to 0. I need to understand why in this situation the ref count was 0 in a way that caused the error.